packages/app-extensions/src/form/reduxForm.js

```diff
@@ -62,19 +63,22 @@ export const transformFieldNameBack = fieldName => (fieldName.replace(/--/g, '.'
 
 export const entityToFormValues = entity => {
   if (!entity || !entity.paths) return {}
-  const result = {}
-  const paths = entity.paths
+
+  const result = {
+    [versionField]: entity.version
+  }
 
   Object.keys(entity.paths).forEach(key => {
-    const field = paths[key]
-    if (field.type === 'entity' || field.type === 'entity-list' || field.type === 'display-expression') {
-      result[transformFieldName(key)] = field.value
-    } else {
-      result[transformFieldName(key)] = field.value ? field.value.value : null
-    }
+    const field = entity.paths[key]
+    const type = field.type
+    const transformedFieldName = transformFieldName(key)
+    const typeValueTransformer = typeValueTransformers[type]
+
+    result[transformedFieldName] = typeValueTransformer
+      ? typeValueTransformer(field.value)
+      : (field.value.value || null)
```

Comment:
there was a null check for `field.value` before (`field.value ? field.value.value : null`). not necessary?